configs/field-types/media.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import Router from 'next/router';
+import { saveMedia, loadMedia } from '../../core/client/api/content';
+import InputLabel from '@material-ui/core/InputLabel';
+import S from 'string';
+import { asField } from 'informed';
+import { makeStyles } from '@material-ui/core/styles';
+import RemoveIcon from '@material-ui/icons/Clear';
+import DownloadIcon from '@material-ui/icons/SaveAlt';
+import { colors } from '@material-ui/core';
+
+const useStyles = makeStyles({
+  wrapper: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  previewWrapper: {
+    margin: 20,
+    height: 180,
+    width: 350,
+    position: 'relative',
+    '&:hover': {
+      opacity: 0.6,
+    },
+  },
+  imgWrapper: {
+    height: '100%',
+    width: '100%',
+  },
+  dropZoneWrapper: {
+    height: 180,
+    width: 350,
+    backgroundColor: colors.grey[200],
+    padding: 10,
+    margin: 20,
+    '&:hover': {
+      backgroundColor: colors.grey[100],
+    },
+  },
+  inputWrapper: {
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flexDirection: 'column',
+    height: '100%',
+    width: '100%',
+    border: '1px dashed silver',
+  },
+  input: {
+    position: 'absolute',
+    height: '100%',
+    width: '100%',
+    opacity: 0,
+  },
+  removeIcon: {
+    position: 'absolute',
+    right: 5,
+    cursor: 'pointer',
+  },
+  downloadIcon: {
+    height: 30,
+    width: 30,
+  },
+  dragInfoLabel: {
+    fontSize: 12,
+    fontWeight: 700,
+    color: 'black',
+  },
+  clickInfoLabel: {
+    fontSize: 10,
+    color: colors.grey[500],
+    paddingBottom: 10,
+  },
+});
+
+export const Renderer = asField(({ fieldState, fieldApi, label, ...props }) => {
+  useEffect(() => {
+    loadMedia(value).then(res => res && setMediaUrl(`/static/${value}`));
+  }, [value])
+  const { value } = fieldState;
+  const { setValue, setTouched } = fieldApi;
+  const { onChange, onBlur, initialValue, forwardedRef, ...rest } = props;
+  const [mediaUrl, setMediaUrl] = useState(null);
```

Comment:
I would move it to the top, before hooks (like useEffect), because you define setState after usage. Yes, it is working, but looks weird